src/Microsoft.ML.Sweeper/Algorithms/NelderMead.cs

```diff
@@ -96,6 +96,11 @@ public NelderMeadSweeper(IHostEnvironment env, Arguments args)
             env.CheckUserArg(args.DeltaExpansion > args.DeltaReflection, nameof(args.DeltaExpansion), "Must be greater than " + nameof(args.DeltaReflection));
             env.CheckUserArg(0 < args.GammaShrink && args.GammaShrink < 1, nameof(args.GammaShrink), "Must be between 0 and 1");
 
+            if (args.FirstBatchSweeper == null)
+            {
+                throw new NullReferenceException("First Batch Sweeper Contains Null Value");
```

Comment:
FYI - for future reference.  You never throw a NullReferenceException yourself.  Only the runtime should throw them.  Instead, you'd want to throw an `ArgumentNullException` here.